### ISSUE TITLE
Add better logging

### ISSUE THREAD

**Issue by pareeohnos** [automatisch/automatisch#1009]
Whilst working on a custom implementation of automatisch, I've stumbled upon a random issue that only crops up once I've deployed it. In my attempts to figure out what was going on, I discovered that automatisch has very limited logging and provides virtually no info into the cause of any exceptions raise. 

For example, when I post the payload for my webhook trigger, I saw one of the following log messages

```
[info]: JOB ID: 9 - FLOW ID: f1fca3f5-e07b-4bee-a1d7-25da2daa97ff has failed to start with Malformed UTF-8 data
```

and

```
[info]: JOB ID: 5 - FLOW ID: f1fca3f5-e07b-4bee-a1d7-25da2daa97ff has failed to start with Unexpected end of JSON input
```

And that's it, nothing else. In the source, I can see there is some integration with Sentry, however I don't have a Sentry account, and I also feel it shouldn't be necessary for have specific error tracking systems just to gain insight into the issues that are going on.

I think that by default the app should be logging out error traces, and at the very least the error log shouldn't be an `info` message but correctly tagged as an `error`

**Comment by farukaydin** [comment]
@pareeohnos, thanks for the feedback. We should definitely improve error logging and the flexibility to change regardless of the app environment, as you mentioned.

The error you mentioned comes from the `bullmq` workers (https://github.com/automatisch/automatisch/blob/main/packages/backend/src/workers/flow.ts#L68), and we actually pass the error message into the logs, but I assume from your feedback that we should pass the whole error including the stack instead of just the message. Is that going to be helpful for you?

Regarding Sentry, we don't want to force our users to use Sentry to gain insights for the errors. It's just for us to use in our cloud (future) version. So we don't have any intention to make error tracking different depending on whether Sentry is integrated or not.

**Comment by pareeohnos** [comment]
@farukaydin Yeah I had to stick some custom logging into that bit of code to gain access to the full stack trace which helped me figure out what issue I was running into. Systems I've dealt with tend to log out full exceptions rather than just the message but that might not be universal, but would certainly be useful if it could at least be configurable. At the very least though I don't think it should be logging the error as an `info` message as it would be very easy to miss any actual errors this way.

Understood about Sentry, that's good to know
